Kernel/Arch/PlatformDriver.cpp

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2024, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Singleton.h>
+#include <Kernel/Arch/PlatformDriver.h>
+
+namespace Kernel {
+
+static Singleton<SpinlockProtected<PlatformDriver::List, LockRank::None>> s_all_instances;
```

Comment:
I'm probably missing something, but what is the purpose of the `Singleton` here, given that we administer this ourselves anyways? Same applies for the `GraphicsConnector` change.